apps/angular/src/app/features/anime/anime-page/anime-page.component.ts

```diff
@@ -1,74 +1,179 @@
+		);
+
+		const scrollToTopAfterChangePageSideEffect$ = this.pagination$.pipe(
+			tap(() => this.scrollToTopPage()),
+		);
+
+		merge(resetPaginationSideEffect$, scrollToTopAfterChangePageSideEffect$)
+			.pipe(untilDestroyed(this))
+			.subscribe();
+	}
+
+	/**
+	 * Create filters form.
+	 * @param filters Initial filters.
+	 */
+	private createFiltersForm(filters: AnimeFilterParams): FiltersForm {
+		return this.formBuilder.group({
+			search: this.formBuilder.control(filters.search),
+			type: this.formBuilder.control(filters.type),
+		});
+	}
+
 	/** Create anime list stream. */
 	private createAnimeListStream(): Observable<Pagination<Anime>> {
-		return this.pagination$.pipe(
-			debounceTime(REQUEST_DEBOUNCE_TIME),
-			map(pagination => this.createParams(pagination)),
-			tap(() => this.isLoading$.next(true)),
-			switchMap(params => this.animeService.getAnime(params)),
-			tap(() => this.isLoading$.next(false)),
-		);
+		return combineLatest([
+			this.filtersForm.valueChanges.pipe(startWith(this.filtersForm.value)),
+			this.pagination$,
+			this.sorting$,
+		])
+			.pipe(
+				debounceTime(REQUEST_DEBOUNCE_TIME),
+				map(([{ search, type }, pagination, sorting]) => this.createAnimeParams(pagination, sorting, search, type)),
+				tap(params => this.setQueryParamsFromAnimeParams(params)),
+				tap(() => this.isLoading$.next(true)),
+				switchMap(params => this.animeService.getAnime(params)),
+				tap(() => this.isLoading$.next(false)),
+			);
+	}
+
+	/**
+	 * Set query params from anime params.
+	 * @param params Anime params.
+	 */
+	private setQueryParamsFromAnimeParams(params: AnimeParams): void {
+		const queryParams = {
+			pageSize: params.pagination.pageSize,
+			pageNumber: params.pagination.pageNumber,
+			field: params.sorting.field,
+			direction: params.sorting.direction,
+			search: params.filters.search,
+			type: params.filters.type.join(','),
+		};
+
+		this.router.navigate([], { queryParams, queryParamsHandling: 'merge' });
 	}
 
 	/**
-	 * Create query params.
-	 * @param pagination Pagination.
+	 * Create anime params.
+	 * @param pagination Pagination: pageNumber and pageSize.
+	 * @param sorting Sorting: field and direction.
+	 * @param search Search.
+	 * @param type Anime type.
 	 */
-	private createParams(pagination: PaginationParams): AnimeParams {
+	private createAnimeParams(pagination: PaginationParams, sorting: Sorting<AnimeSortingField>,
+		search?: string, type?: AnimeType[]): AnimeParams {
 		return {
 			pagination,
-			sorting: defaultParams.sorting,
+			sorting,
+			filters: {
+				search: search ?? defaultParams.filters.search,
+				type: type ?? defaultParams.filters.type,
+			},
+		};
+	}
+
+	/**
+	 * Map query params to anime params.
+	 * @param params Query params.
+	 */
+	private mapQueryParamsToAnimeParams(params: Params): AnimeParams {
+		return {
+			pagination: new PaginationParams({
+				pageSize: +(params['pageSize'] ?? defaultParams.pagination.pageSize),
```

Comment:
Maybe it's debatable, but I believe it would be better to use explicit type casting. Because, I think, we can face with such cases, when this record will be hard to read and when code contains a lot of symbols it can confuse.
```suggestion
				pageSize: Number(params['pageSize'] ?? defaultParams.pagination.pageSize),
```

Reply:
I would even say that if we know we work with integers, we can use `Number.parseInt()` method